### PR TITLE
Fix multiply overflow at src/meta/attribute:1503

### PR DESCRIPTION
**Pull request by anfedotoff**
Fix for #165.

### CODE REVIEW

File: src/meta/attribute.rs [hunk 1]
```diff
@@ -1500,14 +1500,31 @@ impl Preview {
...
-            size: Vec2(width, height),
-            pixel_data,
-        };
+                return Ok(preview);
+            }
+        }
 
-        Ok(preview)
+        return Err(Error::invalid(
+                format!("Overflow while calculating image size \
```
Comment (johannesvollmer) [inline src/meta/attribute.rs:1523]: please mention in the error message that this is about the preview image Attribute, not the actual image

File: src/meta/attribute.rs [hunk 2]
```diff
@@ -1500,14 +1500,31 @@ impl Preview {
...
+                (width: {}, height: {}, components_per_pixel: {}).",
```
Comment (johannesvollmer) [inline src/meta/attribute.rs:1524]: I would remove the components per pixel output, it's always 4. Or maybe rename to bytes per pixel. What do you think?

File: src/meta/attribute.rs [hunk 3]
```diff
@@ -1500,14 +1500,31 @@ impl Preview {
         let width = u32::read(read)? as usize;
         let height = u32::read(read)? as usize;
 
-        let pixel_data = i8::read_vec(read, width * height * components_per_pixel, 1024*1024*4, None, "preview attribute pixel count")?;
+        if let Some(size) = width.checked_mul(height) {
+            if let Some(total_size) = size.checked_mul(components_per_pixel) {
```
Comment (johannesvollmer) [inline src/meta/attribute.rs:1504]: this variable (total_size) never had a nice name in the first place, could you rename it to `byte_count` while you're at it? Would be nice

File: src/meta/attribute.rs [hunk 4]
```diff
@@ -1500,14 +1500,31 @@ impl Preview {
         let width = u32::read(read)? as usize;
         let height = u32::read(read)? as usize;
 
-        let pixel_data = i8::read_vec(read, width * height * components_per_pixel, 1024*1024*4, None, "preview attribute pixel count")?;
+        if let Some(size) = width.checked_mul(height) {
```
Comment (johannesvollmer) [inline src/meta/attribute.rs:1503]: Cool

**Review by johannesvollmer — CHANGES_REQUESTED** [review]

File: src/meta/attribute.rs (hunk 2 above)
Comment (anfedotoff) [reply src/meta/attribute.rs:1524]: It's a good idea to remove. I think we could add a comment before multiplying by four, about what 4 means.

File: src/meta/attribute.rs (hunk 3 above)
Comment (anfedotoff) [reply src/meta/attribute.rs:1504]: `byte_count` sounds good. Maybe change `size`  to `pixel_count`?

Comment (johannesvollmer) [reply src/meta/attribute.rs]: Excellent, indeed

**Comment by johannesvollmer** [comment]
Excellent work, thank you for your contribution! :)

**Review by johannesvollmer — APPROVED** [review]

**Pull request merged**